.editorconfig

```diff
@@ -302,3 +302,38 @@ resharper_blank_lines_before_multiline_statements = 1
 
 resharper_parentheses_non_obvious_operations = arithmetic, multiplicative, equality, relational, additive
 resharper_parentheses_redundancy_style = remove_if_not_clarifies_precedence
+
+dotnet_analyzer_diagnostic.category-roslynator.severity = error
+
+# Remove suffix 'Async' from non-asynchronous method name. Disabled because we like that suffix for now.
+dotnet_diagnostic.RCS1047.severity = none
+
+# Combine 'Enumerable.Where' method chain. It doesn't make it more readable in all cases.
+dotnet_diagnostic.RCS1112.severity = suggestion
+
+# Inline local variable.
+dotnet_diagnostic.RCS1124.severity = suggestion
+
+# Add exception to documentation comment. Nice suggestion, but we don't want to document exceptions for internal code.
+dotnet_diagnostic.RCS1140.severity = suggestion
+
+# Use conditional access. Suggestion because it doesn't always improve readability
+dotnet_diagnostic.RCS1146.severity = suggestion
+
+# Enum should declare explicit values. Disabled because we're not storing them.
+dotnet_diagnostic.RCS1161.severity = none
+
+# Static member in generic type should use a type parameter. Disabled because it's not always applicable.
+dotnet_diagnostic.RCS1158.severity = none
+
+# Add region name to #endregion.
+dotnet_diagnostic.RCS1189.severity = none
+
+# Convert comment to documentation comment. Disabled because it also complains about SMELL/REFACTOR comments
+dotnet_diagnostic.RCS1181.severity = none
+
+# Use Regex instance instead of static method. Disabled because it goes against the Microsoft performance guidelines
+dotnet_diagnostic.RCS1186.severity = none
```

Comment:
It's the opposite. If you don't disable this rule, it will tell you you should _not_ use the static `Regex.IsMatch`.

Reply:
As always "it depends" and the [RCS1186 docs](https://github.com/JosefPihrt/Roslynator/blob/main/docs/analyzers/RCS1186.md) doesn't paint the full picture.

It's trying to tell us that:
* A cache is always beneficial
* it only stores the most recent 15 parsed regexes
* if we store the regex in a (static) member there's no cache misses.

E.g.

```cs
private static readonly Regex regex = new("pattern");

void Foo(List<string> inputs)
{
    foreach (var input in inputs)
    {
        // use the static regex directly
        regex.Match(input); 
    }
}

void Bar(List<string> inputs)
{
    foreach (var input in inputs)
    {
        // check `RegexCache` if the pattern was among the previous 15 used patterns and return the cached Regex.
        // if a cache miss the pattern has to be parsed again.
        Regex.Match("pattern", input); 
    }
}
```

https://source.dot.net/#System.Text.RegularExpressions/System/Text/RegularExpressions/Regex.Match.cs,26
https://source.dot.net/#System.Text.RegularExpressions/System/Text/RegularExpressions/Regex.Cache.cs,31

Reply:
In this case, I don't care. We're only calling this when a test fails and during an equivalency assertion, where the cache is good enough. So up to you to choose.

Reply:
I checked the cases and agree that none of them should be in hot paths.

How about changing the comment to something a long the lines of
```
Use Regex instance instead of static method. Disabled because it's not always worth it.
```